mp4parse/src/lib.rs

```diff
@@ -601,13 +614,58 @@ fn read_moov<T: Read>(f: &mut BMFFBox<T>, context: &mut MediaContext) -> Result<
                 log!("{:?}", mvex);
                 context.mvex = Some(mvex);
             }
+            BoxType::ProtectionSystemSpecificHeaderBox => {
+                let pssh = try!(read_pssh(&mut b));
+                log!("{:?}", pssh);
+                context.psshs.push(pssh);
+            }
             _ => try!(skip_box_content(&mut b)),
         };
         check_parser_state!(b.content);
     }
     Ok(())
 }
 
+fn read_pssh<T: Read>(src: &mut BMFFBox<T>) -> Result<ProtectionSystemSpecificHeaderBox> {
+    let mut box_content = Vec::new();
+    try!(src.read_to_end(&mut box_content));
+
+    let (system_id, kid, data) = {
+        let pssh = &mut Cursor::new(box_content.as_slice());
+
+        let (version, _) = try!(read_fullbox_extra(pssh));
+
+        let system_id = try!(read_buf(pssh, 16));
+
+        let mut kid: Vec<ByteData> = Vec::new();
+        if version > 0 {
+            let mut count = try!(be_i32(pssh));
+            while count > 0 {
+                let item = try!(read_buf(pssh, 16));
+                kid.push(item);
+                count -= 1;
+            }
```

Comment:
I think `for _ in 0..count` would be better than a `while` loop here. It's a little awkward, but there's one less line of code, and `count` doesn't need to be `mut`.